### DISCUSSION TITLE
Error: Cannot read property 'access' of undefined

### DISCUSSION THREAD

**Discussion by personman** [privatenumber/vue-grep#5] [Q&A]
I installed vue-grep on my MacBook and get this error when I try to query. I'm sure I'm missing something simple. Can anyone point me in the right direction? I installed with:  npm i -g vue-grep and vue-grep --version works (and help), but any query throws the error in the title. 

I'm excited to get this working. I was wanting something like this just the other day and I was amazed to stumble on vue-grep today.

versions:

vue-grep: 1.0.0
npm: 7.11.2
node: 10.15.3

**Comment by privatenumber** [comment]
Can you provide the full command you ran and the full error stack trace?

**Reply by personman** [reply]
I tried various query commands:

    vue-grep 'template'
    vue-grep '.subcategory'
    vue-grep 'button'

This is the block of code that gives rise to the error:

    const isCompound = t => t.type === NodeTypes.COMPOUND,
    isRelational = t => [NodeTypes.ADJACENT, NodeTypes.SIBLING, NodeTypes.CHILD, NodeTypes.DESCENDANT].includes(t.type),
    fsExists = async t => fs__namespace.promises.access(t, fs__namespace.constants.F_OK).then(() => !0, () => !1);

fs__namespace.promises is undefined. fs__namespace exists, but does not have a promises attribute. I tried wrapping this block in a try/catch to get more details about the error, but I didn't get anything useful. Please forgive my ignorance.

My project uses webpack 3.12, which might be part of the problem.  Or maybe there's something else strange about my setup. I've tried the queries in multiple vue apps within my project and get the same errors.

**Reply by privatenumber** [reply]
Thanks for the details.

I should've caught it from the first post, but you're using Node.js v10 which has reached end of life earlier this year.

Please use Node v12 or higher.

If your codebase requires v10, you can use [nvm](http://nvm.sh/) to switch to v12 when you want to use vue-grep, and switch back to v10 for anything else.

**Reply by personman** [reply]
Yes, that worked! This tool looks so cool and useful. Thank you for pointing out the problem.  Sorry to hassle you with something that basic. Cheers.

**Reply by privatenumber** [reply]
No worries! Glad you like it :)
